src/cobra/io/web/embl_gems_repository.py

```diff
@@ -0,0 +1,98 @@
+"""
+Provide a concrete implementation of the carveme repository interface.
+"""
+
+
+from io import BytesIO
+
+import httpx
+
+from .abstract_model_repository import AbstractModelRepository
+
+
+def _decode_model_path(model_path):
+    """Decode the model path to EMBL GEMs."""
+    tokens = model_path.split("_")
+    genus = tokens[0]
+
+    directory = genus.lower()
+    alphabet = directory[0]
+
+    return f"{alphabet}/{directory}/{model_path}"
+
+
+class EMBLGems(AbstractModelRepository):
```

Comment:
I think `EMBLGEMs` would also be an acceptable name, although your version is more legible.